### PR TITLE
Feat: Add Secondary Sidebar

### PR DESCRIPTION
**Pull request by TGlide**
## What does this PR do?

Adds a secondary sidebar to certain pages

## Test Plan

Manual

## Related PRs and Issues

N/A

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: src/lib/layout/navigation.svelte [hunk 1]
```diff
@@ -2,81 +2,152 @@
...
 
     $: project = $page.params.project;
-    $: path = `${base}/console/project-${project}`;
+    $: projectPath = `${base}/console/project-${project}`;
+
+    $: secondSideNav = $page.data.secondSideNav;
+    // We need to have this second variable, because we only want narrow
+    // to change automatically if we change from having a second side nav to
+    // not having one, not when the second side nav changes to a different value.
+    $: hasSecondSideNav = !!secondSideNav;
```
Comment (TorstenDittmann) [inline src/lib/layout/navigation.svelte:15]: can we maybe give it a clearer naming? 🤔 

I think something like subNavigation or whatever would work better

File: src/routes/console/project-[project]/databases/database-[database]/+layout.svelte [hunk 2]
```diff
@@ -1,5 +1,28 @@
...
+    }
+</script>
+
 <svelte:head>
     <title>Database - Appwrite</title>
 </svelte:head>
 
 <slot />
+
+<Create bind:showCreate={$showCreate} on:created={handleCreate} />
```
Comment (TorstenDittmann) [inline src/routes/console/project-[project]/databases/database-[database]/+layout.svelte:28]: This is for creating a collection right?

If its now used across layouts, lets give it a clearer name. `showCreateCollection`

File: src/routes/console/project-[project]/databases/database-[database]/collection-[collection]/+layout.ts [hunk 3]
```diff
@@ -16,7 +18,11 @@ export const load: LayoutLoad = async ({ params, parent, depends }) => {
             collection: await sdkForProject.databases.getCollection(
                 params.database,
                 params.collection
-            )
+            ),
+            secondSideNav: SecondSideNav,
```
Comment (TorstenDittmann) [inline src/routes/console/project-[project]/databases/database-[database]/collection-[collection]/+layout.ts:22]: love this solution

File: src/routes/console/project-[project]/databases/database-[database]/collection-[collection]/secondSideNav.svelte [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    import { page } from '$app/stores';
+    import { showCreate } from '../store';
+    import type { PageData } from './[[page]]/$types';
+
+    $: data = $page.data as PageData;
+    $: project = $page.params.project;
+    $: databaseId = $page.params.database;
+    $: collectionId = $page.params.collection;
+
+    $: console.log(data);
```
Comment (TorstenDittmann) [inline src/routes/console/project-[project]/databases/database-[database]/collection-[collection]/secondSideNav.svelte:12]: leftover 👍🏻

**Review by TorstenDittmann — COMMENTED** [review]
looks really good already!

just on mobile view the titles are being shifted around

Bildschirm­foto 2023-03-10 um 22 08 57

File: src/lib/layout/navigation.svelte [hunk 5]
```diff
@@ -2,81 +2,165 @@
...
 
     $: project = $page.params.project;
-    $: path = `${base}/console/project-${project}`;
+    $: projectPath = `${base}/console/project-${project}`;
+
+    $: subNavigation = $page.data.subNavigation;
+    // We need to have this second variable, because we only want narrow
+    // to change automatically if we change from having a second side nav to
+    // not having one, not when the second side nav changes to a different value.
+    $: hasSubNavigation = !!subNavigation;
```
[the remark this answers is not in the document]
Comment (ArmanNik) [reply src/lib/layout/navigation.svelte:15]: Aaaaah I see what you mean now :+1:

**Review by ArmanNik — APPROVED** [review]

**Review by TorstenDittmann — APPROVED** [review]

File: src/lib/layout/navigation.svelte [hunk 6]
```diff
@@ -2,81 +2,165 @@
...
+
+    let narrow = false;
+    $: {
+        narrow = hasSubNavigation;
+    }
+
+    function handleKeyDown(event: KeyboardEvent) {
+        // If Alt + S is pressed
+        if (hasSubNavigation && event.altKey && event.key === 's') {
```
[the remark this answers is not in the document]
Comment (TGlide) [reply src/lib/layout/navigation.svelte]: Fixed

File: src/lib/layout/navigation.svelte [hunk 7]
```diff
@@ -2,81 +2,165 @@
...
+            <div class="side-nav-bottom">
+                <section class="drop-section">
+                    <a
+                        class="drop-button"
+                        href={`${projectPath}/settings`}
+                        on:click={() => trackEvent('click_menu_settings')}
+                        class:is-selected={$page.url.pathname.startsWith(`${projectPath}/settings`)}
+                        title="Settings">
+                        <span class="icon-cog" aria-hidden="true" />
+                        <span class="text">Settings</span>
```
[the remark this answers is not in the document]
Comment (TGlide) [reply src/lib/layout/navigation.svelte:143]: We do, my bad!

**Review by TorstenDittmann — APPROVED** [review]

**Pull request merged**